### PR TITLE
ACM-13623: Add clusterinstance to ACM gather

### PR DESCRIPTION
**Pull request by sakhoury**
Resolves 

**Related Issue:** https://issues.redhat.com/browse/ACM-13623

**Description of Changes:** Adds `ClusterInstance` to ACM must-gather

**What resource(s) are being added:** 
- `clusterinstance.siteconfig.open-cluster-management.io`

**Is this a Hub or Managed cluster change?:**

- [x] Hub Cluster
- [ ] Managed Cluster
- [ ] N/A

**Notes:** Required for SiteConfig operator GA

The SiteConfig operator pods are automatically collected since they deployed under the ACM namespace. The snippet below illustrates the pod logs that are collected for the SiteConfig operator.

```sh
$ ls -lR must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2/
total 32
drwxr-xr-x  3 sakhoury  staff     96 31 Aug 10:22 kube-rbac-proxy
drwxr-xr-x  3 sakhoury  staff     96 31 Aug 10:22 manager
-rwxr-xr-x  1 sakhoury  staff  13939 31 Aug 10:22 siteconfig-controller-manager-7f4c885cdc-9cpm2.yaml

must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2//kube-rbac-proxy:
total 0
drwxr-xr-x  3 sakhoury  staff  96 31 Aug 10:22 kube-rbac-proxy

must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2//kube-rbac-proxy/kube-rbac-proxy:
total 0
drwxr-xr-x  5 sakhoury  staff  160 31 Aug 10:22 logs

must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2//kube-rbac-proxy/kube-rbac-proxy/logs:
total 8
-rwxr-xr-x  1 sakhoury  staff  442 31 Aug 10:22 current.log
-rwxr-xr-x  1 sakhoury  staff    0 31 Aug 10:22 previous.insecure.log
-rwxr-xr-x  1 sakhoury  staff    0 31 Aug 10:22 previous.log

must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2//manager:
total 0
drwxr-xr-x  3 sakhoury  staff  96 31 Aug 10:22 manager

must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2//manager/manager:
total 0
drwxr-xr-x  5 sakhoury  staff  160 31 Aug 10:22 logs

must-gather/namespaces/rhacm/pods/siteconfig-controller-manager-7f4c885cdc-9cpm2//manager/manager/logs:
total 80
-rwxr-xr-x  1 sakhoury  staff  40238 31 Aug 10:22 current.log
-rwxr-xr-x  1 sakhoury  staff      0 31 Aug 10:22 previous.insecure.log
-rwxr-xr-x  1 sakhoury  staff      0 31 Aug 10:22 previous.log
```

### CODE REVIEW

File: collection-scripts/gather_hub_logs [hunk 1]
```diff
@@ -42,6 +42,9 @@ if [ -z $GATHER_HUB_RAN ] || [ $GATHER_HUB_RAN != true ]; then
   run_inspect restores.velero.io --all-namespaces
   run_inspect ansiblejobs.tower.ansible.com --all-namespaces
 
+  # SiteConfig Operator CRs
+  run_inspect clusterinstance.siteconfig.open-cluster-management.io --all-namespaces
```
Comment (dhaiducek) [inline collection-scripts/gather_hub_logs:46]: This should have an `s`, shouldn't it?

ref: https://github.com/stolostron/siteconfig/blob/2bbb072ca7bfe98349aa4a9a9a588e46094580cd/api/v1alpha1/clusterinstance_types.go#L428

```suggestion
  run_inspect clusterinstances.siteconfig.open-cluster-management.io --all-namespaces
```

Comment (sakhoury) [reply collection-scripts/gather_hub_logs]: Updated!

**Review by dhaiducek — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**